src/domain/class/CarMover.js

```diff
@@ -0,0 +1,47 @@
+import Car from './Car';
+
+class CarMover {
+  #moveFunction;
+  #moveDistance;
+
+  constructor(moveFunction, moveDistance) {
+    this.#moveFunction = moveFunction;
+    this.#moveDistance = moveDistance;
+  }
+
+  giveManyTry(car, maxTryCount) {
+    this.#validateTryCount(maxTryCount);
+    this.#giveManyTryRecursive(car, maxTryCount);
+  }
+
+  // #giveManyTryFor(car, maxTryCount) {
+  //   for (let tryCount = 0; tryCount < maxTryCount; tryCount++)
+  //     this.#giveOneTry(car);
+  // }
+
+  // 스택 오버플로우 방지를 위해 2 이상의 카운트가 들어올 경우 분기를 나눠서 실행
+  #giveManyTryRecursive(car, leftCount) {
+    if (leftCount >= 2) {
+      this.#giveManyTryRecursive(car, Math.floor(leftCount / 2));
+      this.#giveManyTryRecursive(car, leftCount - Math.floor(leftCount / 2));
+    }
+    if (leftCount === 1) this.#giveOneTry(car, leftCount - 1);
+  }
+
+  #giveOneTry(car) {
+    if (!(car instanceof Car)) throw new Error('[ERROR] Car가 아님');
+    const canGo = this.#moveFunction();
+    car.grantTry(canGo ? this.#moveDistance : 0);
```

Comment:
소소하지만 이런것도 변수로 분리해줘서 관리하면 가독성에 도움이 될 것 같아요!
```suggestion
    const distance = cango ?  this.#moveDistance : 0
    car.grantTry(distance);
```